src/PowerFlowData.jl

```diff
@@ -1,19 +1,24 @@
-struct PowerFlowData{M <: PNM.PowerNetworkMatrix, N}
+# ! NOTE: bus_magnitude saved as a matrix like for angles and line flows. Is it correct? 
+# ! or just keep a vector?
+
+struct PowerFlowData{M <: PNM.PowerNetworkMatrix, N, S <: Union{String, Char}}
     bus_lookup::Dict{Int, Int}
     branch_lookup::Dict{String, Int}
-    bus_activepower_injection::Vector{Float64}
-    bus_reactivepower_injection::Vector{Float64}
-    bus_activepower_withdrawals::Vector{Float64}
-    bus_reactivepower_withdrawals::Vector{Float64}
+    bus_activepower_injection::Union{Vector{Float64}, Matrix{Float64}}
+    bus_reactivepower_injection::Union{Vector{Float64}, Matrix{Float64}}
```

Comment:
There is no need for this union. This can be just Matrix

Reply:
solved